File: Java/src/test/java/org/codecop/dependencies/d/RestCountriesAPITest.java
package org.codecop.dependencies.d;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Ignore;
import org.junit.Test;

public class RestCountriesAPITest {

    @Ignore("slow")
    @Test
    public void isAmericas() {
        Kingdom us = new Kingdom("US");
        assertTrue(getInstance().isInMagicRealm(us));
    }

    @Ignore("slow")
    @Test
    public void isNotCommonMarket() {
        Kingdom us = new Kingdom("US");
        assertFalse(getInstance().isInAdventurousLands(us));
    }

    @Test
    public void getDistance() {
        Kingdom us = new Kingdom("US");
        assertEquals(8284768, getInstance().distanceTo(us));
    }

    @Ignore("slow")
    @Test
    public void isNotAmericas() {
        Kingdom at = new Kingdom("AT");
        assertFalse(getInstance().isInMagicRealm(at));
    }

    @Ignore("slow")
    @Test
    public void isCommonMarket() {
        Kingdom at = new Kingdom("AT");
        assertTrue(getInstance().isInAdventurousLands(at));
    }

    @Ignore("slow")
    @Test
    public void getNoDistance() {
        Kingdom at = new Kingdom("AT");
        assertEquals(0, getInstance().distanceTo(at));
    }

    private ExplorationGuild getInstance() {
        return ExplorationGuild.ask();
    }

}

File: CSharp/DependencyBreakingTests/D/RestCountriesAPITest.cs
using Org.Codecop.Dependencies.D;
using Xunit;

namespace Org.Codecop.Dependencies.D.Tests
{
    public class RestCountriesAPITest
    {
        [Fact(Skip = "slow")]
        public void IsAmericas()
        {
            Country us = new Country("US");
            Assert.True(GetInstance().IsInAmericas(us));
        }

        [Fact(Skip = "slow")]
        public void IsNotCommonMarket()
        {
            Country us = new Country("US");
            Assert.False(GetInstance().IsInCommonMarket(us));
        }

        [Fact]
        public void GetDistance()
        {
            Country us = new Country("US");
            Assert.Equal(8284768, GetInstance().DistanceTo(us));
        }

        [Fact(Skip = "slow")]
        public void IsNotAmericas()
        {
            Country at = new Country("AT");
            Assert.False(GetInstance().IsInAmericas(at));
        }

        [Fact(Skip = "slow")]
        public void IsCommonMarket()
        {
            Country at = new Country("AT");
            Assert.True(GetInstance().IsInCommonMarket(at));
        }

        [Fact(Skip = "slow")]
        public void GetNoDistance()
        {
            Country at = new Country("AT");
            Assert.Equal(0, GetInstance().DistanceTo(at));
        }

        private RestCountriesAPI GetInstance()
        {
            return RestCountriesAPI.GetInstance();
        }
    }

}
